src/coreclr/jit/instr.cpp

```diff
@@ -795,11 +813,104 @@ CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
             assert(op->OperIsHWIntrinsic());
 
 #if defined(FEATURE_HW_INTRINSICS)
-            GenTreeHWIntrinsic* hwintrinsic = op->AsHWIntrinsic();
-            NamedIntrinsic      intrinsicId = hwintrinsic->GetHWIntrinsicId();
-
+            GenTreeHWIntrinsic* hwintrinsic  = op->AsHWIntrinsic();
+            NamedIntrinsic      intrinsicId  = hwintrinsic->GetHWIntrinsicId();
+            var_types           simdBaseType = hwintrinsic->GetSimdBaseType();
             switch (intrinsicId)
             {
+                case NI_AVX_BroadcastScalarToVector128:
+                case NI_AVX_BroadcastScalarToVector256:
+                {
+                    // we have the assumption that AVX_BroadcastScalarToVector*
+                    // only take the memory address as the operand.
+                    assert(hwintrinsic->isContained());
+                    assert(hwintrinsic->OperIsMemoryLoad());
+                    assert(hwintrinsic->GetOperandCount() == 1);
+                    assert(varTypeIsFloating(simdBaseType));
+                    GenTree* broadcastScalar = hwintrinsic->Op(1);
+                    assert(broadcastScalar->isContained());
+                    if (broadcastScalar->OperIs(GT_LCL_ADDR))
```

Comment:
I'm concerned that we could be missing handling here. We're effectively able to encounter anything that an indirection could have as its address, so that is at least `LCL_ADDR`, `CNS_INT`, and `LEA`. But it could also potentially include `CLS_VAR_ADDR`

I expect there's toerhs we could encounter as well, but those are at least what `ContainCheckIndir` currently covers

Reply:
Unlike the two comments below, we can't handle this recursively since the `IND` logic is a bit special and only handled in `genCodeForIndir`/`emitInsLoadInd` and not more generally.

It might be desirable, long term, to have such a shared function, but at least for now we should ensure all the same cases are being handled so we don't hit any asserts in production.

Reply:
So for `NI_AVX_BroadcastScalarToVector*`, we need to handle the those cases: `LCL_VAR`, `LCL_ADDR`, `CLS_VAR_ADDR`, `CNS_INT` and `LEA`. There are `STRUCT` and `IND_REQ_ADDR_IN_REG` but these 2 seems irrelevant, is that right?

For the mentioned case, I imagine the cases for  `LCL_ADDR`, `CLS_VAR_ADDR`, `CNS_INT` might be the same, `LCL_VAR` need to be handled separately, then what about `LEA`?